baseplate/diagnostics/tracing.py

```diff
@@ -0,0 +1,226 @@
+"""Components for processing Baseplate spans for service request tracing."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import calendar
+import json
+import logging
+import socket
+import time
+from datetime import datetime
+
+import gevent
+import requests
+
+from ..core import BaseplateObserver, SpanObserver, TraceInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+# Span annotation types
+ANNOTATIONS = {
+    'CLIENT_SEND': 'cs',
+    'CLIENT_RECEIVE': 'cr',
+    'SERVER_SEND': 'ss',
+    'SERVER_RECEIVE': 'sr',
+}
+
+
+def current_epoch_microseconds():
+    """Return current UTC time since epoch in microseconds."""
+    epoch_ts = datetime.utcfromtimestamp(0)
+    return int((datetime.utcnow() - epoch_ts).
+               total_seconds() * 1000 * 1000)
```

Comment:
This is a good idea, thanks for the additional functions! I'll add a followup PR for a time-utility library so we can provide consistency across baseplate services.